From 1cfb4715288845ebc55ad664421b48b32de9599c Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Fri, 7 Aug 2020 14:38:43 +0000
Subject: [PATCH 015/267] Update syscall lists for Linux 5.8.

Linux 5.8 has one new syscall, faccessat2.  Update syscall-names.list
and regenerate the arch-syscall.h headers with build-many-glibcs.py
update-syscalls.

Tested with build-many-glibcs.py.
---
 sysdeps/unix/sysv/linux/aarch64/arch-syscall.h           | 1 +
 sysdeps/unix/sysv/linux/alpha/arch-syscall.h             | 1 +
 sysdeps/unix/sysv/linux/arc/arch-syscall.h               | 1 +
 sysdeps/unix/sysv/linux/arm/arch-syscall.h               | 1 +
 sysdeps/unix/sysv/linux/csky/arch-syscall.h              | 1 +
 sysdeps/unix/sysv/linux/hppa/arch-syscall.h              | 1 +
 sysdeps/unix/sysv/linux/i386/arch-syscall.h              | 1 +
 sysdeps/unix/sysv/linux/ia64/arch-syscall.h              | 1 +
 sysdeps/unix/sysv/linux/m68k/arch-syscall.h              | 1 +
 sysdeps/unix/sysv/linux/microblaze/arch-syscall.h        | 1 +
 sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h       | 1 +
 sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h   | 1 +
 sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h   | 1 +
 sysdeps/unix/sysv/linux/nios2/arch-syscall.h             | 1 +
 sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h | 1 +
 sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h | 1 +
 sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h        | 1 +
 sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h      | 1 +
 sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h      | 1 +
 sysdeps/unix/sysv/linux/sh/arch-syscall.h                | 1 +
 sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h     | 1 +
 sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h     | 1 +
 sysdeps/unix/sysv/linux/syscall-names.list               | 5 +++--
 sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h         | 1 +
 sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h        | 1 +
 25 files changed, 27 insertions(+), 2 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h b/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h
index 81767d6777..ebbb5e13c3 100644
--- a/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h
@@ -33,6 +33,7 @@
 #define __NR_exit 93
 #define __NR_exit_group 94
 #define __NR_faccessat 48
+#define __NR_faccessat2 439
 #define __NR_fadvise64 223
 #define __NR_fallocate 47
 #define __NR_fanotify_init 262
diff --git a/sysdeps/unix/sysv/linux/alpha/arch-syscall.h b/sysdeps/unix/sysv/linux/alpha/arch-syscall.h
index 97db195b7c..f0744f241b 100644
--- a/sysdeps/unix/sysv/linux/alpha/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/alpha/arch-syscall.h
@@ -45,6 +45,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 405
 #define __NR_faccessat 462
+#define __NR_faccessat2 549
 #define __NR_fadvise64 413
 #define __NR_fallocate 480
 #define __NR_fanotify_init 494
diff --git a/sysdeps/unix/sysv/linux/arc/arch-syscall.h b/sysdeps/unix/sysv/linux/arc/arch-syscall.h
index 58cd927da8..591ff88869 100644
--- a/sysdeps/unix/sysv/linux/arc/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/arc/arch-syscall.h
@@ -37,6 +37,7 @@
 #define __NR_exit 93
 #define __NR_exit_group 94
 #define __NR_faccessat 48
+#define __NR_faccessat2 439
 #define __NR_fadvise64_64 223
 #define __NR_fallocate 47
 #define __NR_fanotify_init 262
diff --git a/sysdeps/unix/sysv/linux/arm/arch-syscall.h b/sysdeps/unix/sysv/linux/arm/arch-syscall.h
index d13eeedf4d..42ea0fda16 100644
--- a/sysdeps/unix/sysv/linux/arm/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/arm/arch-syscall.h
@@ -54,6 +54,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 248
 #define __NR_faccessat 334
+#define __NR_faccessat2 439
 #define __NR_fallocate 352
 #define __NR_fanotify_init 367
 #define __NR_fanotify_mark 368
diff --git a/sysdeps/unix/sysv/linux/csky/arch-syscall.h b/sysdeps/unix/sysv/linux/csky/arch-syscall.h
index 6273d1bd20..64dc411b9d 100644
--- a/sysdeps/unix/sysv/linux/csky/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/csky/arch-syscall.h
@@ -39,6 +39,7 @@
 #define __NR_exit 93
 #define __NR_exit_group 94
 #define __NR_faccessat 48
+#define __NR_faccessat2 439
 #define __NR_fadvise64_64 223
 #define __NR_fallocate 47
 #define __NR_fanotify_init 262
diff --git a/sysdeps/unix/sysv/linux/hppa/arch-syscall.h b/sysdeps/unix/sysv/linux/hppa/arch-syscall.h
index 80b7278d12..56e3bad12b 100644
--- a/sysdeps/unix/sysv/linux/hppa/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/hppa/arch-syscall.h
@@ -51,6 +51,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 222
 #define __NR_faccessat 287
+#define __NR_faccessat2 439
 #define __NR_fadvise64_64 236
 #define __NR_fallocate 305
 #define __NR_fanotify_init 322
diff --git a/sysdeps/unix/sysv/linux/i386/arch-syscall.h b/sysdeps/unix/sysv/linux/i386/arch-syscall.h
index 6669c1bb56..cc554ca5b7 100644
--- a/sysdeps/unix/sysv/linux/i386/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/i386/arch-syscall.h
@@ -55,6 +55,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 252
 #define __NR_faccessat 307
+#define __NR_faccessat2 439
 #define __NR_fadvise64 250
 #define __NR_fadvise64_64 272
 #define __NR_fallocate 324
diff --git a/sysdeps/unix/sysv/linux/ia64/arch-syscall.h b/sysdeps/unix/sysv/linux/ia64/arch-syscall.h
index 60ec095fde..50405e4d04 100644
--- a/sysdeps/unix/sysv/linux/ia64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/ia64/arch-syscall.h
@@ -44,6 +44,7 @@
 #define __NR_exit 1025
 #define __NR_exit_group 1236
 #define __NR_faccessat 1293
+#define __NR_faccessat2 1463
 #define __NR_fadvise64 1234
 #define __NR_fallocate 1303
 #define __NR_fanotify_init 1323
diff --git a/sysdeps/unix/sysv/linux/m68k/arch-syscall.h b/sysdeps/unix/sysv/linux/m68k/arch-syscall.h
index 2edecb1b72..49ceaa8b3a 100644
--- a/sysdeps/unix/sysv/linux/m68k/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/m68k/arch-syscall.h
@@ -55,6 +55,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 247
 #define __NR_faccessat 300
+#define __NR_faccessat2 439
 #define __NR_fadvise64 246
 #define __NR_fadvise64_64 267
 #define __NR_fallocate 320
diff --git a/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h b/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h
index d72845c1f5..2bc15c6021 100644
--- a/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h
@@ -55,6 +55,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 252
 #define __NR_faccessat 307
+#define __NR_faccessat2 439
 #define __NR_fadvise64 250
 #define __NR_fadvise64_64 272
 #define __NR_fallocate 324
diff --git a/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h b/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h
index 29217a1304..57e0fe0b3b 100644
--- a/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h
@@ -56,6 +56,7 @@
 #define __NR_exit 4001
 #define __NR_exit_group 4246
 #define __NR_faccessat 4300
+#define __NR_faccessat2 4439
 #define __NR_fadvise64 4254
 #define __NR_fallocate 4320
 #define __NR_fanotify_init 4336
diff --git a/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h b/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h
index 26996a61c8..202329795c 100644
--- a/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h
@@ -53,6 +53,7 @@
 #define __NR_exit 6058
 #define __NR_exit_group 6205
 #define __NR_faccessat 6263
+#define __NR_faccessat2 6439
 #define __NR_fadvise64 6216
 #define __NR_fallocate 6283
 #define __NR_fanotify_init 6300
diff --git a/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h b/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h
index f046a77351..8e1dd0ca94 100644
--- a/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h
@@ -48,6 +48,7 @@
 #define __NR_exit 5058
 #define __NR_exit_group 5205
 #define __NR_faccessat 5259
+#define __NR_faccessat2 5439
 #define __NR_fadvise64 5215
 #define __NR_fallocate 5279
 #define __NR_fanotify_init 5295
diff --git a/sysdeps/unix/sysv/linux/nios2/arch-syscall.h b/sysdeps/unix/sysv/linux/nios2/arch-syscall.h
index cf85d11b4a..d107c5e34c 100644
--- a/sysdeps/unix/sysv/linux/nios2/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/nios2/arch-syscall.h
@@ -38,6 +38,7 @@
 #define __NR_exit 93
 #define __NR_exit_group 94
 #define __NR_faccessat 48
+#define __NR_faccessat2 439
 #define __NR_fadvise64_64 223
 #define __NR_fallocate 47
 #define __NR_fanotify_init 262
diff --git a/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h b/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h
index 90ff02b3f1..2ef6c88541 100644
--- a/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h
@@ -54,6 +54,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 234
 #define __NR_faccessat 298
+#define __NR_faccessat2 439
 #define __NR_fadvise64 233
 #define __NR_fadvise64_64 254
 #define __NR_fallocate 309
diff --git a/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h b/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h
index 311a560d86..130e23514d 100644
--- a/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h
@@ -49,6 +49,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 234
 #define __NR_faccessat 298
+#define __NR_faccessat2 439
 #define __NR_fadvise64 233
 #define __NR_fallocate 309
 #define __NR_fanotify_init 323
diff --git a/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h b/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h
index 1f90f25891..385722c6b3 100644
--- a/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h
@@ -33,6 +33,7 @@
 #define __NR_exit 93
 #define __NR_exit_group 94
 #define __NR_faccessat 48
+#define __NR_faccessat2 439
 #define __NR_fadvise64 223
 #define __NR_fallocate 47
 #define __NR_fanotify_init 262
diff --git a/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h b/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h
index 92210b6137..18a8ed1b00 100644
--- a/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h
@@ -53,6 +53,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 248
 #define __NR_faccessat 300
+#define __NR_faccessat2 439
 #define __NR_fadvise64 253
 #define __NR_fadvise64_64 264
 #define __NR_fallocate 314
diff --git a/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h b/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h
index af031e9734..d59248e2b2 100644
--- a/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h
@@ -45,6 +45,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 248
 #define __NR_faccessat 300
+#define __NR_faccessat2 439
 #define __NR_fadvise64 253
 #define __NR_fallocate 314
 #define __NR_fanotify_init 332
diff --git a/sysdeps/unix/sysv/linux/sh/arch-syscall.h b/sysdeps/unix/sysv/linux/sh/arch-syscall.h
index e554ad25bd..e69906c492 100644
--- a/sysdeps/unix/sysv/linux/sh/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/sh/arch-syscall.h
@@ -52,6 +52,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 252
 #define __NR_faccessat 307
+#define __NR_faccessat2 439
 #define __NR_fadvise64 250
 #define __NR_fadvise64_64 272
 #define __NR_fallocate 324
diff --git a/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h b/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h
index cb93f94259..ee186fa5bd 100644
--- a/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h
@@ -54,6 +54,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 188
 #define __NR_faccessat 296
+#define __NR_faccessat2 439
 #define __NR_fadvise64 209
 #define __NR_fadvise64_64 210
 #define __NR_fallocate 314
diff --git a/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h b/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h
index 6f108ffd6b..7ce4150d6d 100644
--- a/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h
@@ -48,6 +48,7 @@
 #define __NR_exit 1
 #define __NR_exit_group 188
 #define __NR_faccessat 296
+#define __NR_faccessat2 439
 #define __NR_fadvise64 209
 #define __NR_fadvise64_64 210
 #define __NR_fallocate 314
diff --git a/sysdeps/unix/sysv/linux/syscall-names.list b/sysdeps/unix/sysv/linux/syscall-names.list
index 15dec5b98f..a462318ecf 100644
--- a/sysdeps/unix/sysv/linux/syscall-names.list
+++ b/sysdeps/unix/sysv/linux/syscall-names.list
@@ -21,8 +21,8 @@
 # This file can list all potential system calls.  The names are only
 # used if the installed kernel headers also provide them.
 
-# The list of system calls is current as of Linux 5.7.
-kernel 5.7
+# The list of system calls is current as of Linux 5.8.
+kernel 5.8
 
 FAST_atomic_update
 FAST_cmpxchg
@@ -105,6 +105,7 @@ execveat
 exit
 exit_group
 faccessat
+faccessat2
 fadvise64
 fadvise64_64
 fallocate
diff --git a/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h b/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h
index d22e5b2da9..8dc0062a53 100644
--- a/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h
@@ -48,6 +48,7 @@
 #define __NR_exit 60
 #define __NR_exit_group 231
 #define __NR_faccessat 269
+#define __NR_faccessat2 439
 #define __NR_fadvise64 221
 #define __NR_fallocate 285
 #define __NR_fanotify_init 300
diff --git a/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h b/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h
index 9a6f4f16d2..1a701c1472 100644
--- a/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h
@@ -44,6 +44,7 @@
 #define __NR_exit 1073741884
 #define __NR_exit_group 1073742055
 #define __NR_faccessat 1073742093
+#define __NR_faccessat2 1073742263
 #define __NR_fadvise64 1073742045
 #define __NR_fallocate 1073742109
 #define __NR_fanotify_init 1073742124
-- 
2.29.0

